api-report/task-manager.api.md

```diff
@@ -29,10 +29,17 @@ export interface ITaskManager extends ISharedObject<ITaskManagerEvents> {
 
 // @public
 export interface ITaskManagerEvents extends ISharedObjectEvents {
-    (event: "assigned" | "completed" | "lost", listener: (taskId: string) => void): any;
+    // Warning: (ae-forgotten-export) The symbol "TaskEventListener" needs to be exported by the entry point index.d.ts
+    //
+    // @eventProperty
+    (event: "assigned", listener: TaskEventListener): any;
+    // @eventProperty
+    (event: "completed", listener: TaskEventListener): any;
+    // @eventProperty
+    (event: "lost", listener: TaskEventListener): any;
 }
 
-// @public
+// @public @sealed
```

Comment:
@scottn12 just want to double check that changing this class to sealed makes sense. Are consumers expected to sub-class this?

Reply:
Taking the question 1 step further: Would it make sense (as a breaking change on `next`) to make this class private to the package, and expose a factory function for creating an `ITaskManagerr` instead?

Reply:
We don't intend to have customers to use `TaskManager` directly, however I think completely hiding it away could cause issues, especially when setting up an instance of TaskManager. For example, we use it in this data migration example: 

https://github.com/microsoft/FluidFramework/blob/af627a48fab9cea7bb7f14ce4a4886882499df78/examples/utils/example-utils/src/migrationTool/migrationTool.ts#LL158C62-L158C62

cc @ChumpChief - you may want to weigh in here

Reply:
Hiding away the class makes forwards / backwards compat significantly easier, but discoverability can suffer a bit in larger API surfaces. For our DDS packages, though, it would have my vote. And I wholeheartedly agree about being consistent across our suite here - just wanted to take the advantage to raise the topic again 😉